src/components/BottomSheet/BottomSheet.tsx

```diff
@@ -0,0 +1,212 @@
+import {View, Text, Dimensions, StyleSheet, TextInput, ViewProps} from 'react-native'
+import React, {forwardRef, memo, useCallback, useImperativeHandle, useMemo, useState} from 'react'
+import styled from 'styled-components'
+import BottomSheetBackdropContainer from './BottomSheetBackdropContainer'
+import BottomSheetContainer from './BottomSheetContainer/BottomSheetContainer'
+import Animated, {
+  runOnJS,
+  runOnUI,
+  useAnimatedKeyboard,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  useWorkletCallback,
+  withSpring,
+  withTiming,
+} from 'react-native-reanimated'
+import type {BottomSheetMethods, BottomSheetProps} from './types'
+import BottomSheetContent from './BottomSheetContent/BottomSheetContent'
+import {Gesture, GestureDetector, GestureHandlerRootView} from 'react-native-gesture-handler'
+import {
+  BOTTOM_SHEET_STATE,
+  INITIAL_CONTAINER_HEIGHT,
+  INITIAL_CONTAINER_OFFSET,
+  SCREEN_HEIGHT,
+  WINDOW_HEIGHT,
+} from './constants'
+import {useNormalizedSnapPoints} from './utils'
+import {BottomSheetProvider} from './contexts/BottomSheetProvider'
+import {styles} from './styles'
+
+const {width, height} = Dimensions.get('window')
+
+const Container = styled(View)(() => ({
+  width,
+  height,
+  position: 'absolute',
+}))
+
+const BottomSheetComponent = (
```

Comment:
Where is type for Props

Reply:
it's here 

<img width="872" alt="image" src="https://github.com/saigontechnology/rn-base-component/assets/117077260/32aac6d9-1cd3-42c9-b54e-6bd5f0cb2999">